typescript/sdk/src/ism/EvmIsmModule.ts

```diff
@@ -1,89 +1,727 @@
-import { Address, ProtocolType, rootLogger } from '@hyperlane-xyz/utils';
+import { ethers } from 'ethers';
+import { Logger } from 'pino';
 
-import { HyperlaneContracts } from '../contracts/types.js';
+import {
+  DefaultFallbackRoutingIsm__factory,
+  DomainRoutingIsm,
+  DomainRoutingIsmFactory__factory,
+  DomainRoutingIsm__factory,
+  IAggregationIsm,
+  IAggregationIsm__factory,
+  IInterchainSecurityModule__factory,
+  IMultisigIsm,
+  IMultisigIsm__factory,
+  IRoutingIsm,
+  MailboxClient__factory,
+  OPStackIsm__factory,
+  Ownable__factory,
+  PausableIsm__factory,
+  StaticAddressSetFactory,
+  StaticThresholdAddressSetFactory,
+  TestIsm__factory,
+  TrustedRelayerIsm__factory,
+} from '@hyperlane-xyz/core';
+import {
+  Address,
+  Domain,
+  ProtocolType,
+  assert,
+  eqAddress,
+  objFilter,
+  rootLogger,
+} from '@hyperlane-xyz/utils';
+
+import { attachAndConnectContracts } from '../contracts/contracts.js';
+import { HyperlaneAddresses, HyperlaneContracts } from '../contracts/types.js';
 import {
   HyperlaneModule,
   HyperlaneModuleArgs,
 } from '../core/AbstractHyperlaneModule.js';
 import { HyperlaneDeployer } from '../deploy/HyperlaneDeployer.js';
-import { ProxyFactoryFactories } from '../deploy/contracts.js';
+import {
+  ProxyFactoryFactories,
+  proxyFactoryFactories,
+} from '../deploy/contracts.js';
+import { extractOwnerAddress } from '../deploy/types.js';
 import { MultiProvider } from '../providers/MultiProvider.js';
-import { EthersV5Transaction } from '../providers/ProviderType.js';
-import { ChainNameOrId } from '../types.js';
+import {
+  AnnotatedEthersV5Transaction,
+  createAnnotatedEthersV5Transaction,
+} from '../providers/ProviderType.js';
+import { ChainMap, ChainName, ChainNameOrId } from '../types.js';
+import { findMatchingLogEvents } from '../utils/logUtils.js';
 
-import { EvmIsmCreator } from './EvmIsmCreator.js';
 import { EvmIsmReader } from './EvmIsmReader.js';
-import { IsmConfig } from './types.js';
+import {
+  AggregationIsmConfig,
+  DeployedIsm,
+  DeployedIsmType,
+  IsmConfig,
+  IsmType,
+  MultisigIsmConfig,
+  RoutingIsmConfig,
+  RoutingIsmDelta,
+} from './types.js';
+import { moduleMatchesConfig, routingModuleDelta } from './utils.js';
+
+type ExtraArgs = {
+  deployedIsm: Address;
+  mailbox: Address;
+};
```

Comment:
is `deployedIsm` similar to `WithAddress<IsmConfig>`
can we dedupe these interfaces

Reply:
this is just an address, i was actually considering removing the config arg from the abstract module